pkg/helm/release/manager_factory.go

```diff
@@ -182,15 +252,29 @@ func getReleaseName(storageBackend *storage.Storage, crChartName string, cr *uns
 	// If a release name with the CR name exists, but the release's chart is
 	// different than the chart managed by this operator, return an error
 	// because something else created the existing release.
-	existingChartName := history[0].GetChart().GetMetadata().GetName()
+	if history[0].Chart == nil {
+		return "", fmt.Errorf("could not find chart metadata in release with name %q", releaseName)
+	}
+	existingChartName := history[0].Chart.Name()
 	if existingChartName != crChartName {
 		return "", fmt.Errorf("duplicate release name: found existing release with name %q for chart %q", releaseName, existingChartName)
 	}
 
 	return releaseName, nil
 }
 
-func releaseHistory(storageBackend *storage.Storage, releaseName string) ([]*rpb.Release, bool, error) {
+func releaseHistoryV2(storageBackend *v2storage.Storage, releaseName string) ([]*helmreleasev2.Release, bool, error) {
+	releaseHistory, err := storageBackend.History(releaseName)
+	if err != nil {
+		if notFoundErr(err) {
+			return nil, false, nil
+		}
+		return nil, false, err
+	}
+	return releaseHistory, len(releaseHistory) > 0, nil
+}
+
+func releaseHistoryV3(storageBackend *v3storage.Storage, releaseName string) ([]*helmreleasev3.Release, bool, error) {
```

Comment:
Why is required have releaseHistoryV2 and releaseHistoryV3? Will we keep the compatibility with old versions?
Should it still work on with helms v2?

Reply:
This is necessary to handle automatically migrating existing Helm v2 releases to Helm v3. The idea here is that a user with existing CRs and an existing Helm v2-based operator can upgrade to the new Helm 3-based operator and the new operator will handle the transition seamlessly.